src/main/java/edu/njit/jerse/daikonplusplus/inv/Invariant.java

```diff
@@ -0,0 +1,64 @@
+package edu.njit.jerse.daikonplusplus.inv;
+
+import java.util.Objects;
+
+/**
+ * Represents a candidate or inferred invariant as a Java-style expression string, along with its
+ * validation status.
+ */
+public class Invariant {
+
+  /** The actual invariant expression, e.g., "x > 0". */
+  private final String expression;
+
+  /** Whether this invariant has been falsified by dynamic evidence. */
+  private boolean falsified;
+
+  /**
+   * Constructs a new invariant with the given expression. By default, it is assumed to be valid
+   * (not falsified).
+   *
+   * @param expression The Java-style Boolean expression representing the invariant.
+   */
+  public Invariant(String expression) {
+    this.expression = expression;
+    this.falsified = false;
```

Comment:
I would use an initializer expression for this rather than setting it in the constructor, in case we want to add another constructor in the future.